internal/binder/function/funcs_agg.go

```diff
@@ -487,7 +509,7 @@ func dedup(r []interface{}, col []interface{}, all bool) (interface{}, error) {
 	}
 	if !all {
 		if len(result) == 0 {
-			return nil, nil
+			return []interface{}{}, nil
```

Comment:
The second argument 'all' means whether to reutrn the full array or just a single event. Here, we need to return a single event -> nil